### PR TITLE
fix(flub release): Account for RC release branch names

### PR DESCRIPTION
**Pull request by tylerbutler**
The calculation of release branch names did not properly account for the RC releases. I fixed that and added a unit test.

The whole generateReleaseBranchName function could use some refactoring to be simpler, but that's a larger change and I don't think it's important enough to do now.

### CODE REVIEW

File: build-tools/packages/build-cli/test/library/branches.test.ts [hunk 1]
```diff
@@ -129,6 +129,12 @@ describe("generateReleaseBranchName", () => {
 		assert.equal(actual, expected);
 	});
 
+	it("Fluid RC version scheme", () => {
+		const actual = generateReleaseBranchName(MonoRepoKind.Client, "2.0.0-rc.1.0.0");
```
Comment (markfields) [inline build-tools/packages/build-cli/test/library/branches.test.ts:133]: Also test that --branchCheck accepts RC branches?

Comment (tylerbutler) [reply build-tools/packages/build-cli/test/library/branches.test.ts:133]: That's not a test that we can easily write. Doing so would require us to mock the whole repo, becuuse the command queries the repo itself to check the current branch.

However, unit testing these functions is 99% of what's needed. Most of the command logic is just glue between individually testable functions.

**Comment by tylerbutler** [comment]
> Note that #18828 is still pending.

Approved and merged.

**Review by alexvy86 — APPROVED** [review]
Seems fine as far as I can tell

**Comment by tylerbutler** [comment]
Thanks for the review, @alexvy86!

**Pull request merged**